Server/src/main/resources/static/js/editor.js

```diff
@@ -175,6 +175,55 @@ function parseDWInstruction(line, result, currentLine) {
     }
 }
 
+function isLabel(text, result) {
+    if (result === undefined) { return false; }
+    
+    for (var i = 0; i < result.labels.length; i++) {
+        if (text === result.labels[i]) {
+            return true;
+        }
+    }
+}
+
+function isImmediateOp(text, result) {
+    return (isLabel(text, result) 
+        || !isNaN(Number(text)) 
+        && Number(text) === Math.floor(Number(text)) 
+        && text.indexOf("o") === -1
+        && text.indexOf("0e") !== 0);
+}
+
+function isMemoryOp(text) {
+    return text.startsWith("[") && text.endsWith("]");
+}
+
+function isRegisterOp(text) {
+    return new RegExp('^(a|b|c|d|x|y|bp|sp)$').test(text.toLowerCase());
+}
+
+function getOffsetOperandType(text) {
```

Comment:
`result` is missing here as an argument name. Javascript does not check if you function signature is correct. It just supplies the calling arguments to the function and any arguments not supplied have the value `undefined`. If you supply more arguments than the function takes it just ignores those arguments.

Reply:
Oh my goodness, I don't know how I didn't catch that